mempool/reactor.go

```diff
@@ -271,19 +281,20 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 // included.
 // WARNING: mutates next!
 func (memR *Reactor) txs(next *clist.CElement, peerID uint16, peerHeight int64) [][]byte {
-	batch := make([][]byte, 0)
+	batch := make([][]byte, 0, memR.config.MaxBatchBytes)
```

Comment:
This can't be right? We're allocating a number of byte slices equal to `MaxBatchBytes`, so with e.g. 4096 we're allocating 4096 separate byte slices.